lib/core/ResolveCache.js

```diff
@@ -11,6 +11,10 @@ var semver = require('../util/semver');
 var readJson = require('../util/readJson');
 var copy = require('../util/copy');
 
+function _md5(source) {
```

Comment:
i wouldn't create a new function just for this, you can add a line in line 48 like

```js
// avoid multiple folders for same dependency because different hash
var sourceLowerCase = source.toLowerCase();
var sourceId = md5(sourceLowerCase);
```
This way we only modify the code base in one place and address the problem.